### PR TITLE
feat: introduce `tls.disable_http2_alpn()` function

### PR DESCRIPTION
**Pull request by oowl**

### CODE REVIEW

File: src/ngx_http_lua_kong_ssl.c [hunk 1]
```diff
@@ -175,6 +176,48 @@ ngx_http_lua_kong_get_upstream_ssl_verify(ngx_http_request_t *r,
     return ngx_lua_kong_ssl_get_upstream_ssl_verify(&ctx->ssl_ctx, proxy_ssl_verify);
 }
 
+ngx_flag_t
+ngx_lua_kong_ssl_enable_http2_alpn(ngx_ssl_connection_t *ssl,
```
Comment (windmgc) [inline src/ngx_http_lua_kong_ssl.c:180]:
```suggestion
ngx_http_lua_kong_ssl_get_http2_alpn_enabled(ngx_ssl_connection_t *ssl,
```
Propose another name with the necessary "_http_" prefix and replace the verb to show the real meaning

File: lualib/resty/kong/tls.lua [hunk 2]
```diff
@@ -314,6 +318,24 @@ do
 
         error("unknown return code: " .. tostring(ret))
     end
+
+    function _M.disable_http2_alpn()
```
Comment (windmgc) [inline lualib/resty/kong/tls.lua:322]: Could you add docs for this function inside `README.md`?

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -5,7 +5,7 @@ on:
   push:
 
 env:
-  KONG_VERSION: master
+  KONG_VERSION: disable-h2-alpn-re
```
Comment (tzssangglass) [inline .github/workflows/tests.yml:8]: for test?

Comment (oowl) [reply .github/workflows/tests.yml:8]: for build with patch https://github.com/Kong/kong/pull/13709 in CI test action.

**Review by Water-Melon — APPROVED** [review]

File: lualib/resty/kong/tls.lua [hunk 4]
```diff
@@ -59,6 +61,7 @@ if subsystem == "http" then
         int depth);
     int ngx_http_lua_kong_ffi_get_socket_ssl(ngx_http_lua_socket_tcp_upstream_t *u,
         void **ssl_conn);
+    int ngx_http_lua_ffi_ssl_disable_http2_alpn(ngx_http_request_t *r, char **err);
```
Comment (fffonion) [inline lualib/resty/kong/tls.lua:64]: remove ssl_ and name it `ngx_http_lua_ffi_disable_http2_alpn` to be consistent with others

File: src/ngx_http_lua_kong_module.h [hunk 5]
```diff
@@ -36,5 +36,8 @@ ngx_flag_t
 ngx_http_lua_kong_get_upstream_ssl_verify(ngx_http_request_t *r,
     ngx_flag_t proxy_ssl_verify);
 
+ngx_flag_t
+ngx_http_lua_kong_ssl_get_http2_alpn_enabled(ngx_ssl_connection_t *ssl,
```
Comment (fffonion) [inline src/ngx_http_lua_kong_module.h:40]: this function seems not used?

Comment (oowl) [reply src/ngx_http_lua_kong_module.h:40]: https://github.com/Kong/kong/pull/13709/files#diff-4242fc0c9bdeb3525f6463c1cec41f726e360b0b8b897be3abab6d83863c649fR21 used in kong side openresty patch

**Review by fffonion — APPROVED** [review]

**Pull request merged**